d3-selection/index.d.ts

```diff
@@ -845,7 +878,7 @@ interface Selection<GElement extends BaseType, Datum, PElement extends BaseType,
     /**
      * Return the first (non-null) element in this selection. If the selection is empty, returns null.
      */
-    node(): GElement | null;
+    node(): GElement;
```

Comment:
I have to test out an alternative  approach, now that two of the minor "blindspots" are mostly removed.
By that I mean, that the `lib.es6.d.ts` is itself improving its `strictNullChecking` and the TS bug that impacted `this` has by and large been fixed.

The gist of it is the following, let the return type of top-level `select` and sub-selection `select` add `null` to the return type, i.e. `Selection<GElement, ..., ..., ...>` becomes `Selection<GElement | null, ..., ..., ...>`. By implication, `this` context of the selection and return type of the `node()` method consistently include the possible `null`.